File: 2022/ruby/template.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'stringio'
require_relative 'test'

def parse_io_line(line)
  line
end

def parse_io(io)
  lines = io.readlines chomp: true
  lines.map { |line| parse_io_line line }
end

def part1(io)
  parse_io io
end

def part2(io)
  parse_io io
end

example = <<~EOF
EOF
Test.example StringIO.open(example) { |io| part1 io }, nil
# Test.example StringIO.open(example) { |io| part2 io }, nil

input = "#{File.basename(__FILE__, '.rb')}.txt"
Test.solution File.open(input) { |io| part1 io }, nil
# Test.solution File.open(input) { |io| part2 io }, nil

File: 2023/lua/template.lua
#!/usr/bin/env lua

local utils = require('utils')
local testing = require('testing')

local function get_input_filename()
  return debug.getinfo(1,'S').source:sub(2):match("[^/]*.lua$"):gsub('.lua', '.txt')
end

local function parse_io_line(line)
  return line
end

local function parse_io(file)
  local lines = {}
  for line in file:lines() do
    local d = parse_io_line(line)
    table.insert(lines, d)
  end

  return lines
end

local function part1(file)
  parse_io(file)
end

local function part2(file)
  parse_io(file)
end

local example = [[
]]

local file = utils.make_tmpfile(example)
testing.example(part1(file), nil)
-- testing.example(part2(file), nil)

file = assert(io.open(get_input_filename()))
testing.solution(part1(file), nil)
-- testing.solution(part2(file), nil)
